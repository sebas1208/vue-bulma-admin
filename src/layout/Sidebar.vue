<template>
    <div class="cdm-sidebar" :class="{'is-mobile': isMobile && isMobileMenuOpen}">
        <nav class="navbar is-dark is-spaced has-border-bottom" :class="{'is-spaced-mobile': isMobile}" role="navigation"
            aria-label="main navigation">

            <div class="navbar-brand">
                <a href="https://bulma.io">
                    <img src="https://bulma.io/images/made-with-bulma--dark.png" alt="Made with Bulma" width="180" height="100">
                </a>
            </div>
        </nav>
        <section class="is-fullheight cmd-sidebar-menu">
            <aside class="menu">
                <ul class="menu-list">
                    <li class="menu-item">
                        <router-link to="/dashboard">
                            <span class="icon is-large">
                                <i class="fas fa-2x fa-home">
                                </i>
                            </span>
                            Dashboard
                        </router-link>
                    </li>
                    <li class="menu-item">
                        <router-link to="/tables">
                            <span class="icon is-large">
                                <i class="fas fa-2x fa-table"></i>
                            </span>
                            Tables
                        </router-link>
                    </li>
                    <li class="menu-item">
                        <router-link to="/component">
                            <span class="icon is-large">
                                <i class="fas fa-2x fa-book"></i>
                            </span>
                            Components
                        </router-link>
                    </li>
                </ul>
            </aside>
        </section>
    </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from "vue-property-decorator";

@Component
export default class Sidebar extends Vue {

    @Prop(Boolean) isMobile!: boolean;
    @Prop(Boolean) isMobileMenuOpen!: boolean;

    constructor() {
        super();
    }
}
</script>
<style lang="scss" scoped>
@import '@/_variables.scss';

.cdm-sidebar {
    background-color: $dark;
    width: $sidebar-width;
    padding-left: 12px;
    padding-right: 12px;
    height: 100%;
}

.menu-item {
    margin-bottom: 20px;
}

.is-mobile {
    position: fixed;
    display: block;
    right: 0;
    left: auto;
    z-index: 10;
    visibility: visible;
    overflow-y: visible;
    border-top: none;
    text-align: left;
}

.cmd-sidebar-menu {
    padding-top: 24px;
}
</style>
