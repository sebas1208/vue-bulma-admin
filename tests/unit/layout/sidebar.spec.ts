import { shallowMount, mount, Wrapper } from "@vue/test-utils";
import Sidebar from "@/layout/Sidebar.vue";


describe("Sidebar Test", () => {
    let wrapper: Wrapper<Sidebar>;

    beforeEach(() => {
       wrapper = shallowMount(Sidebar);
    });

    it("Test Sidebar Mount", () => {
        expect(wrapper.html()).toContain("General");
    });

    it("SnapshotTest", () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});
