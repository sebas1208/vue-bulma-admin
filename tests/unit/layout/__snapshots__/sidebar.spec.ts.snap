// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Test SnapshotTest 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar is-black"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
        href="https://bulma.io"
      >
        <img
          height="28"
          src="https://bulma.io/images/bulma-logo.png"
          width="112"
        />
      </a>
    </div>
  </nav>
   
  <br />
   
  <aside
    class="menu"
  >
    <p
      class="menu-label"
    >
      
            General
        
    </p>
     
    <ul
      class="menu-list"
    >
      <li>
        <a>
          Dashboard
        </a>
      </li>
       
      <li>
        <a>
          Customers
        </a>
      </li>
    </ul>
     
    <p
      class="menu-label"
    >
      
            Administration
        
    </p>
     
    <ul
      class="menu-list"
    >
      <li>
        <a>
          Team Settings
        </a>
      </li>
       
      <li>
        <a
          class="is-active"
        >
          Manage Your Team
        </a>
         
        <ul>
          <li>
            <a>
              Members
            </a>
          </li>
           
          <li>
            <a>
              Plugins
            </a>
          </li>
           
          <li>
            <a>
              Add a member
            </a>
          </li>
        </ul>
      </li>
       
      <li>
        <a>
          Invitations
        </a>
      </li>
       
      <li>
        <a>
          Cloud Storage Environment Settings
        </a>
      </li>
       
      <li>
        <a>
          Authentication
        </a>
      </li>
    </ul>
     
    <p
      class="menu-label"
    >
      
            Transactions
        
    </p>
     
    <ul
      class="menu-list"
    >
      <li>
        <a>
          Payments
        </a>
      </li>
       
      <li>
        <a>
          Transfers
        </a>
      </li>
       
      <li>
        <a>
          Balance
        </a>
      </li>
    </ul>
  </aside>
</div>
`;
